### ISSUE TITLE
Unable to support TS 5.0 tsconfig extends array

### ISSUE THREAD

**Issue by unional** [withastro/astro#6918]
### What version of `astro` are you using?

2.3.1

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

pnpm

### What operating system are you using?

Windows

### What browser are you using?

Chrome

### Describe the Bug

`tsconfig-resolver` does not support `extends` with array, which is a feature in TypeScript 5.0

```ts
// tsconfig.json
{
  "extends: ["astro/tsconfigs/strict", "another-config"]
}
```

It errors with:

```sh
The "path" argument must be of type string. Received an instance of Array
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-3dbiun?file=tsconfig.json

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by bluwy** [comment]
Not sure if there's anything we can do since it needs to be supported by https://github.com/ifiokjr/tsconfig-resolver, which doesn't seem to have recent activity. Another option is to use https://github.com/dominikg/tsconfck or https://github.com/aleclarson/vite-tsconfig-paths

**Comment by Princesseuh** [comment]
We should probably migrate to `tsconfck` anyway, `tsconfig-resolver` is unmaintained and has some issues we needed workaround for in the past

**Comment by Princesseuh** [comment]
Blocked by https://github.com/dominikg/tsconfck/issues/87, [I started working on a PR for it in this fork](https://github.com/Princesseuh/tsconfck), but since this is kinda low priority for us, I'll circle back to it later unless someone is interested in tackling it.

**Comment by MichaelDeBoey** [comment]
@Princesseuh I would be interested to get your `jsconfig.json` support merged into `tsfonck` for the reasons mentioned in https://github.com/dominikg/tsconfck/issues/87 👀

**Comment by firxworx** [comment]
After having wasted a chunk of time on an obtuse error where I could get Nx + Astro 3 to build on my workstation and not with Dockerfile and finally tracing it to this... My issue: I extended multiple tsconfigs (!!)

I haven't checked but it sort of smells to me like there might be a bit of "find an npm package for anything and everything" at play with that one where dependencies get created on a single maintainer packages that appear abandoned to save a few lines of code.

Taking a peek at tsconfig-resolver's source (https://www.npmjs.com/package/tsconfig-resolver?activeTab=code), it seems like a good chunk of tsconfig-resolver effectively duplicates what Astro already has in its own codebase as a core part of its functionality plus in existing Astro packages I already see other deps on well-used libraries like fast-glob etc.

Perhaps this sort of thing could be more standardized between the official plugins.

https://www.reddit.com/media?url=https%3A%2F%2Fi.redd.it%2Ftfugj4n3l6ez.png
(the classic Heaviest Objects In The Universe meme re node_modules) :D

**Comment by Princesseuh** [comment]
> After having wasted a chunk of time on an obtuse error where I could get Nx + Astro 3 to build on my workstation and not with Dockerfile and finally tracing it to this... My issue: I extended multiple tsconfigs (!!)
> 
> I haven't checked but it sort of smells to me like there might be a bit of "find an npm package for anything and everything" at play with that one where dependencies get created on a single maintainer packages that appear abandoned to save a few lines of code.
> 
> Taking a peek at tsconfig-resolver's source (https://www.npmjs.com/package/tsconfig-resolver?activeTab=code), it seems like a good chunk of tsconfig-resolver effectively duplicates what Astro already has in its own codebase as a core part of its functionality plus in existing Astro packages I already see other deps on well-used libraries like fast-glob etc.
> 
> Perhaps this sort of thing could be more standardized between the official plugins.
> 
> https://www.reddit.com/media?url=https%3A%2F%2Fi.redd.it%2Ftfugj4n3l6ez.png
> (the classic Heaviest Objects In The Universe meme re node_modules) :D
> 

I wish it was the case, but resolving and parsing `js/tsconfig.json` fully is actually not that easy 😅 The logic around `extends` especially is a bit tricky since it's a custom Node-like resolution specific to TypeScript

There's a reason every framework tend to use a library / depend on TypeScript's native methods for that. If you don't implement it to the T, you risk having a different understanding of the TypeScript config in different contexts.

**Comment by so1ve** [comment]
https://github.com/privatenumber/get-tsconfig 👀
